src/Entity/DatasetSubmission.php

```diff
@@ -2600,6 +2600,27 @@ public function getErdappDatasetLink(): ?DatasetLink
         return null;
     }
 
+    /**
+     * Getter for the NCEI url.
+     *
+     * @return string|null
+     */
+    public function getNceiUrl(): ?string
+    {
+        $datasetLinks = $this->getDatasetLinks()->filter(function (DatasetLink $datasetLink) {
+            return $datasetLink->getName() === DatasetLink::LINK_NAME_CODES["ncei"]["name"];
+        });
+
+        if ($datasetLinks->count() > 0) {
+            $nceiLink = $datasetLinks->first();
+        }
+
+        if ($nceiLink instanceof DatasetLink) {
+            return $nceiLink->getUrl();
```

Comment:
Not sure if we need a two-step function, like erdap, but this is ok. We can change it if we need specifics.